web-registry/src/components/organisms/CreditActivityTable.tsx

```diff
@@ -0,0 +1,247 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import { makeStyles } from '@mui/styles';
+import Table from '@mui/material/Table';
+import TableBody from '@mui/material/TableBody';
+import TableHead from '@mui/material/TableHead';
+import TableRow from '@mui/material/TableRow';
+import TableFooter from '@mui/material/TableFooter';
+import Box from '@mui/material/Box';
+import dayjs from 'dayjs';
+import relativeTime from 'dayjs/plugin/relativeTime';
+import { Dictionary } from 'lodash';
+
+import {
+  StyledTableContainer,
+  StyledTableCell,
+  StyledTableRow,
+  StyledTableSortLabel,
+  formatNumber,
+} from 'web-components/lib/components/table';
+import { useTablePagination } from 'web-components/lib/components/table/useTablePagination';
+import {
+  getComparator,
+  stableSort,
+  Order,
+} from 'web-components/lib/components/table/sort';
+import { Label } from 'web-components/lib/components/label';
+import OutlinedButton from 'web-components/lib/components/buttons/OutlinedButton';
+import { Theme } from 'web-components/lib/theme/muiTheme';
+
+import { truncate } from '../../lib/wallet';
+import { ledgerRestUri } from '../../ledger';
+import { getHashUrl } from '../../lib/block-explorer';
+import { getEcocreditTxs } from '../../lib/ecocredit';
+
+dayjs.extend(relativeTime);
+
+interface TxRowData {
+  date: string | Date;
+  txhash: string;
+  messages: string;
+  height: number;
+  txUrl: string;
+}
+
+interface HeadCell {
+  id: keyof TxRowData;
+  label: string;
+  numeric: boolean;
+  wrap?: boolean;
+}
+
+const headCells: HeadCell[] = [
+  { id: 'date', numeric: false, label: 'date' },
+  { id: 'txhash', numeric: false, label: 'hash' },
+  { id: 'messages', numeric: false, label: 'messages' },
+  { id: 'height', numeric: true, label: 'block height' },
+  { id: 'txUrl', numeric: false, label: '' },
+];
+
+const useStyles = makeStyles((theme: Theme) => ({
+  label: {
+    color: theme.palette.info.dark,
+    fontSize: theme.typography.pxToRem(12),
+  },
+}));
+
+const ROWS_PER_PAGE_OPTIONS = { options: [5, 10, 20, 50], default: 10 };
+
+const READABLE_NAMES: Dictionary<string> = {
+  '/regen.ecocredit.v1alpha1.MsgSend': 'send',
+  '/regen.ecocredit.v1alpha1.MsgRetire': 'retire',
+  '/regen.ecocredit.v1alpha1.MsgCreateClass': 'create credit class',
+  '/regen.ecocredit.v1alpha1.MsgCancel': 'cancel',
+  '/regen.ecocredit.v1alpha1.MsgCreateBatch': 'create batch',
```

Comment:
the messages type names could eventually be pulled from regen-js directly (once MsgService impl is merged and released), could you add a TODO?

Reply:
done